element_session/export/nwb.py

```diff
@@ -0,0 +1,130 @@
+import datetime
+from uuid import uuid4
+
+import numpy as np
+from pynwb import NWBFile
+from pynwb.file import Subject
+
+from . import session
+
+try:
+    import element_animal
+    HAVE_ELEMENT_ANIMAL = True
+except:
+    HAVE_ELEMENT_ANIMAL = False
+try:
+    import element_lab
+    HAVE_ELEMENT_LAB = True
+except:
+    HAVE_ELEMENT_LAB = False
+
+
+def session_to_nwb(session_key: dict, subject_id=None):
+    """Gather session- and subject-level metadata and use it to create an NWBFile.
+
+    Parameters
+    ----------
+    session_key: dict,
+        e.g.,
+        {
+            'subject': 'subject5',
+            'session_datetime': datetime.datetime(2020, 5, 12, 4, 13, 7)
+        }
+    subject_id: str, optional
+        Indicate subject_id if it cannot be inferred
+
+    Returns
+    -------
+    pynwb.NWBFile
+
+    mappings:
+        Session.KEY -> NWBFile.session_id
+        Sesion.session_note -> NWBFile.session_description
+        Session.session_datetime -> NWBFile.session_start_time
+        SessionExperimenter.experimenter -> NWBFile.experimenter
```

Comment:
This may be a bit confusing since the dot notation is not used to access attributes in DataJoint. I would suggest alternative notation to list fields, e.g.
```suggestion
        Session :: KEY -> NWBFile.session_id
        Sesion :: session_note -> NWBFile.session_description
        Session :: session_datetime -> NWBFile.session_start_time
        SessionExperimenter :: experimenter -> NWBFile.experimenter
```